src/flamenco/capture/fd_solcap_proto.h

```diff
@@ -2,178 +2,230 @@
 #define HEADER_fd_src_flamenco_capture_fd_solcap_proto_h
 
 #include "../fd_flamenco_base.h"
+#include "../types/fd_types.h"
+#include <sys/types.h>
+#include <stdlib.h>
+#include <stdbool.h>
+
+#include <bits/stdint-uintn.h>
 
 /* fd_solcap_proto defines the capture data format "solcap".
 
-   solcap is a portable file format for capturing Solana runtime data
-   suitable for replay and debugging.  It is laid out as follows:
-
-     File Header
-     File Protobuf Object
-     Chunk 0
-       Chunk Header
-       Chunk Protobuf Object
-       Data ...
-     Chunk 1
-       Chunk Header
-       Chunk Protobuf Object
-       Data ...
-     Chunk N ...
-
-   The file header briefly describes the file's content type and points
-   to the first chunk.  Additional metadata, such as slot bounds, are
-   stored in a Protobuf blob following the header.  Currently, the
-   following content types are implemented:
-
-     SOLCAP_V1_BANK:  Bank pre-image, version 0
-                      (assumed to only contain SOLCAP_V1_BANK chunks)
-
-   Capture content is divided into variable-length chunks. Each chunk
-   contains a fixed-size binary header containing type and length
-   information. Following the header is a serialized Protobuf object
-   with chunk-specific information.
-
-   Typically, readers sequentially read in chunks, loading one chunk
-   into memory at a time.  Within a chunk, data structures are laid out
-   in arbitrary order which requires random access.  Random access out-
-   side of chunks is rarely required.  Readers should ignore chunks of
-   unknown content type.
-
-   Furthermore:
-   - Byte order is little endian
-   - There should be no gaps between slots
-   - Suffix `_foff` ("file offset") refers to an offset from the
-     beginning of a stream
-   - Suffix `_coff` ("chunk offset") refers to an offset from the first
-     byte of the header of the current chunk
-
-   Why a mix of C structs and Protobufs?  We prefer the use of Protobuf
-   to easily support additions to the schema.  Fixed-size/fixed-offset
-   structures are only used to support navigating the file. */
-
-/* TODO Pending features:
-   - Fork support
-   - Compression
-   - Chunk Table */
-
-/* FD_SOLCAP_V1_FILE_MAGIC identifies a solcap version 0 file. */
-
-#define FD_SOLCAP_V1_FILE_MAGIC (0x806fe7581b1da4b7UL)
-
-/* fd_solcap_fhdr_t is the file header of a capture file. */
-
-struct fd_solcap_fhdr {
-  /* 0x00 */ ulong magic;        /* ==FD_SOLCAP_V1_NULL_MAGIC */
-  /* 0x08 */ ulong chunk0_foff;  /* Offset of first chunk from begin of stream */
-
-  /* Metadata;  Protobuf fd_solcap_FileMeta */
-  /* 0x10 */ uint meta_sz;
-  /* 0x14 */ uint _pad14[3];
+   .solcap is a portable file format for capturing Solana runtime data
+   suitable for replay and debugging. It is laid out below:
```

Comment:
solcap is not a file format. This is just a pcapng file capturing solcap packets. 
Important to make this distinction so it's clear to the reader that this format will work with existing tools like Wireshark.